src/main/java/me/karakelley/http/server/ConnectionHandler.java

```diff
@@ -0,0 +1,41 @@
+package me.karakelley.http.server;
+
+import me.karakelley.http.exceptions.InvalidRequestException;
+import me.karakelley.http.Request;
+import me.karakelley.http.RequestParser;
+import me.karakelley.http.Response;
+import me.karakelley.http.responses.BadRequest;
+import me.karakelley.http.handlers.Handler;
+import me.karakelley.http.utility.BufferedLineReader;
+import me.karakelley.http.utility.LineReader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.DataOutputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.Socket;
+
+public class ConnectionHandler {
```

Comment:
I think this was another great decision pulling this out from `HttpServer`—this class now defines the _what_, and `HttpServer` handles the _when_ (i.e. asynchronously upon receiving a request).